net/http/authn/authn.go

```diff
@@ -0,0 +1,100 @@
+package authn
+
+import (
+	"context"
+	"encoding/hex"
+	"net"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"chain/core/accesstoken"
+	"chain/errors"
+	"chain/net/authz"
+)
+
+const tokenExpiry = time.Minute * 5
+
+type API struct {
+	Tokens           *accesstoken.CredentialStore
```

Comment:
How would you feel if I only added that TODO? I feel like moving the CredentialStore code is the better solution, but I'm not sure I want to move so much code right now.